### PR TITLE
DB: Setup flow pattern functions and runs read fucntions

### PR DESCRIPTION
**Pull request by benedeki**
* Setup function for flow patterns
* Some selected read functions for runs - both segmentation and flow based.

### CODE REVIEW

File: database/src/main/flow_patterns/insert_additional_data.sql [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+--
+-- Returns:
+--      status                  - Status code
+--      status_text             - Status text
+--      id_fp_additional_data   - id of the additional data pattern entry
+--      key_fp_flow             - id of the flow the additional data pattern were added into
+--
+-- Status codes:
+--      11                  - Additional data entry created
+--      41                  - Flow not found
```
Comment (dk1844) [inline database/src/main/flow_patterns/insert_additional_data.sql:47]: Is there a reason why this is `41 Flow not found` (as in type `41 - master record not found`), while other calls respond by `40 Flow not found` (as in type `40 - general not found`)?

**Review by dk1844 — APPROVED** [review]
Except for seemingly 40/41 error codes being both used for apparently a similar situation (please correct me if I am wrong), I have not found anything suspicious (LGTM, just read the code)

File: database/src/main/flow_patterns/insert_additional_data.sql (hunk 1 above)
Comment (benedeki) [reply database/src/main/flow_patterns/insert_additional_data.sql:47]: In this case master record is not found in an attempt to add data from master-detail relationship.
But it's totally possible to use 40 as well, if the contact between DB and app would consider in more useful 😉

Comment (benedeki) [reply database/src/main/flow_patterns/insert_additional_data.sql:47]: In this case master record is not found in an attempt to add data from master-detail relationship.
But it's totally possible to use 40 as well, if the contact between DB and app would consider in more useful 😉

**Pull request merged**